From 24e7af1e78c9401c4a9b23e40252e187c3e4ba55 Mon Sep 17 00:00:00 2001
From: Xiyue Deng <manphiz@gmail.com>
Date: Wed, 11 Sep 2024 15:22:01 -0700
Subject: [PATCH 1/2] Add patch to suppress startup warnings on newer wxWidgets
 versions

* Newer wxWidgets added more checks for wxSizer and the old code base
has not migrated yet.
* Added "wxSizerFlags::DisableConsistencyChecks()" to disable the
checks.
* Use newer API to get rid of a warning.
---
 debian/patches/series                         |  1 +
 .../patches/suppress-startup-warnings.patch   | 51 +++++++++++++++++++
 2 files changed, 52 insertions(+)
 create mode 100644 debian/patches/suppress-startup-warnings.patch

diff --git a/debian/patches/series b/debian/patches/series
index bc373f1..2c3e6a0 100644
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -27,3 +27,4 @@ update-ffms2-doindexing2.patch
 update-ffms2-x11-none.patch
 update-ffms2-log-levels.patch
 fix_boost183.patch
+suppress-startup-warnings.patch
diff --git a/debian/patches/suppress-startup-warnings.patch b/debian/patches/suppress-startup-warnings.patch
new file mode 100644
index 0000000..55919ae
--- /dev/null
+++ b/debian/patches/suppress-startup-warnings.patch
@@ -0,0 +1,51 @@
+From bb124f7cced4a117de6adc3caceffed827050e52 Mon Sep 17 00:00:00 2001
+From: Xiyue Deng <manphiz@gmail.com>
+Date: Wed, 11 Sep 2024 16:01:12 -0700
+Subject: [PATCH] Suppress startup warnings
+Forwarded: not-needed
+
+* Newer wxWidgets added more sizer checks and caused a few startup
+warnings.
+* This patches suppresses the warnings by disabling consistency check
+and using newer API.
+* Warning text:
+,----
+| ./src/common/sizer.cpp(2288): assert "CheckSizerFlags(!((flags) & (wxALIGN_RIGHT | wxALIGN_CENTRE_HORIZONTAL | wxALIGN_BOTTOM | wxALIGN_CENTRE_VERTICAL)))" failed in DoInsert(): wxALIGN_RIGHT | wxALIGN_CENTRE_HORIZONTAL | wxALIGN_BOTTOM | wxALIGN_CENTRE_VERTICAL will be ignored in this sizer: wxEXPAND overrides alignment flags in box sizers
+|
+| DO NOT PANIC !!
+|
+| If you're an end user running a program not developed by you, please ignore this message, it is harmless, and please try reporting the problem to the program developers.
+|
+| You may also set WXSUPPRESS_SIZER_FLAGS_CHECK environment variable to suppress all such checks when running this program.
+|
+| If you're the developer, simply remove this flag from your code to avoid getting this message. You can also call wxSizerFlags::DisableConsistencyChecks() to globally disable all such checks, but this is strongly not recommended.
+`----
+---
+ src/frame_main.cpp | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/src/frame_main.cpp b/src/frame_main.cpp
+index ba107d6..50a9242 100644
+--- a/src/frame_main.cpp
++++ b/src/frame_main.cpp
+@@ -98,6 +98,7 @@ FrameMain::FrameMain()
+ , context(agi::make_unique<agi::Context>())
+ {
+ 	StartupLog("Entering FrameMain constructor");
++	wxSizerFlags::DisableConsistencyChecks();
+ 
+ #ifdef __WXGTK__
+ 	// XXX HACK XXX
+@@ -242,8 +243,7 @@ void FrameMain::SetDisplayMode(int video, int audio) {
+ 	TopSizer->Show(videoBox, showVideo, true);
+ 	ToolsSizer->Show(audioBox, showAudio, true);
+ 
+-	MainSizer->CalcMin();
+-	MainSizer->RecalcSizes();
++	MainSizer->RepositionChildren(MainSizer->CalcMin());
+ 	MainSizer->Layout();
+ 	Layout();
+ 
+-- 
+2.39.2
+
-- 
2.39.2

